File: desert/continuous-alchemy.lua
loop
    local e = self:findNearest(self:findEnemies())
    if e then
        local i = self:findNearest(self:findItems())
        if not i then
            self:say("Give me a drink!")
        elseif i.type ~= "poison" then
            self:moveXY(i.pos.x, i.pos.y)
            self:moveXY(34,47)
        end
    end
end

File: desert/continuous-alchemy.clj
(while true
  (def es (.findEnemies this))
  (when (seq es)
    (def is (.findItems this))
    (if (seq is)
      (when (not= (.type (first is)) "poison")
        (.moveXY this (.x (.pos (first is))) (.y (.pos (first is))))
        (.moveXY this 34 47))
      (.say this "Give me a drink!"))))
